### PR TITLE
v0.49.0-beta.1

### PR DESCRIPTION
**Pull request by ryanhamley**
## Launch Checklist

 - [x] briefly describe the changes in this PR
    Update `CHANGELOG` and `package.json`

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,3 +1,33 @@
+## 0.49.0
+September 6, 2018
+
+## ⚠️ Breaking changes
+* Use `client{Height/Width}` instead of `offset{Height/Width}` for map canvas sizing ([#6848](https://github.com/mapbox/mapbox-gl-js/issues/6848), fixed by [#7128](https://github.com/mapbox/mapbox-gl-js/pull/7128))
+
+## 🐛 Bug fixes
+* Fix inadvertent purging of feature state by requiring manual feature state reset ([#7141](https://github.com/mapbox/mapbox-gl-js/pull/7141))
```
Comment (jfirebaugh) [inline CHANGELOG.md]: This was included in 0.48.0.

Comment (mollymerp) [inline CHANGELOG.md]: Oh I think this actually was cherry picked into the final 0.49.0 release

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,3 +1,33 @@
...
+* Fix Top Issues list for mapbox-gl-js ([#7108](https://github.com/mapbox/mapbox-gl-js/issues/7108), fixed by [#7112](https://github.com/mapbox/mapbox-gl-js/pull/7112))
+* Fix bug in which symbols with `icon-allow-overlap: true, text-allow-overlap: true, text-optional: false` would show icons when they shouldn't ([#7041](https://github.com/mapbox/mapbox-gl-js/pull/7041))
+* Fix bug in `Map#flyTo` where the map wouldn't stop on an exact zoom level when the final zoom value is not the one requested by `flyTo` (#7222) ([#7223](https://github.com/mapbox/mapbox-gl-js/pull/7223)) (h/t [benoitbzl](https://github.com/benoitbzl))
+* Keep map centered on the center point of a multi-touch gesture when zooming (#6722) ([#7191](https://github.com/mapbox/mapbox-gl-js/pull/7191)) (h/t [pakastin](https://github.com/pakastin))
+* Update the style-spec's old `gl-style-migrate` script to include conversion of legacy functions and filters to their expression equivalents ([#6927](https://github.com/mapbox/mapbox-gl-js/issues/6927), fixed by [#7095](https://github.com/mapbox/mapbox-gl-js/pull/7095))
+* Fix `icon-size` for small data-driven values ([#7125](c))
```
Comment (jfirebaugh) [inline CHANGELOG.md]: Missing a URL here.

File: CHANGELOG.md [hunk 3]
```diff
@@ -1,3 +1,33 @@
+## 0.49.0
+September 6, 2018
+
+## ⚠️ Breaking changes
+* Use `client{Height/Width}` instead of `offset{Height/Width}` for map canvas sizing ([#6848](https://github.com/mapbox/mapbox-gl-js/issues/6848), fixed by [#7128](https://github.com/mapbox/mapbox-gl-js/pull/7128))
+
+## 🐛 Bug fixes
+* Fix inadvertent purging of feature state by requiring manual feature state reset ([#7141](https://github.com/mapbox/mapbox-gl-js/pull/7141))
+* Fix Top Issues list for mapbox-gl-js ([#7108](https://github.com/mapbox/mapbox-gl-js/issues/7108), fixed by [#7112](https://github.com/mapbox/mapbox-gl-js/pull/7112))
```
Comment (mollymerp) [inline CHANGELOG.md]: add a link to https://mapbox.github.io/top-issues/#!mapbox/mapbox-gl-js ?

File: CHANGELOG.md [hunk 4]
```diff
@@ -1,3 +1,33 @@
...
+
+## ✨ Features and improvements
+Performance updates:
+  * Improve time to first render by updating how feature ID maps are transferred to the main thread ([#7110](https://github.com/mapbox/mapbox-gl-js/issues/7110), fixed by [#7132](https://github.com/mapbox/mapbox-gl-js/pull/7132))
+  * Reduce size of JSON transmitted from worker thread to main thread ([#7124](https://github.com/mapbox/mapbox-gl-js/pull/7124))
+  * Improve image/glyph atlas packing algorithm ([#7171](https://github.com/mapbox/mapbox-gl-js/pull/7171))
+  * Use murmur hash on symbol instance keys to reduce worker transfer costs ([#7127](https://github.com/mapbox/mapbox-gl-js/pull/7127))
+* Add GL state management for uniforms ([#6018](https://github.com/mapbox/mapbox-gl-js/pull/6018))
+* Add `symbol-z-order` symbol layout property to style spec ([#7219](https://github.com/mapbox/mapbox-gl-js/pull/7219))
+* Implement data-driven styling support for ``*-pattern properties` ([#6289](https://github.com/mapbox/mapbox-gl-js/pull/6289))
```
Comment (jfirebaugh) [inline CHANGELOG.md]: Stray `

File: CHANGELOG.md [hunk 5]
```diff
@@ -1,3 +1,33 @@
...
+September 6, 2018
+
+## ⚠️ Breaking changes
+* Use `client{Height/Width}` instead of `offset{Height/Width}` for map canvas sizing ([#6848](https://github.com/mapbox/mapbox-gl-js/issues/6848), fixed by [#7128](https://github.com/mapbox/mapbox-gl-js/pull/7128))
+
+## 🐛 Bug fixes
+* Fix inadvertent purging of feature state by requiring manual feature state reset ([#7141](https://github.com/mapbox/mapbox-gl-js/pull/7141))
+* Fix Top Issues list for mapbox-gl-js ([#7108](https://github.com/mapbox/mapbox-gl-js/issues/7108), fixed by [#7112](https://github.com/mapbox/mapbox-gl-js/pull/7112))
+* Fix bug in which symbols with `icon-allow-overlap: true, text-allow-overlap: true, text-optional: false` would show icons when they shouldn't ([#7041](https://github.com/mapbox/mapbox-gl-js/pull/7041))
+* Fix bug in `Map#flyTo` where the map wouldn't stop on an exact zoom level when the final zoom value is not the one requested by `flyTo` (#7222) ([#7223](https://github.com/mapbox/mapbox-gl-js/pull/7223)) (h/t [benoitbzl](https://github.com/benoitbzl))
```
Comment (mollymerp) [inline CHANGELOG.md]: I think this should just be `Fix bug where the map would not stop at the exact zoom level requested by Map#FlyTo`

**Pull request merged**